### PR TITLE
Enable TF and Keras auto signature logging

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Enable signature logging by default on autologging. Remove backwards compatible logic handling for 1.x.

## How is this patch tested?

Existing tests.

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Tensorflow and Keras autologging will now log signatures by default.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [X] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by WeichenXu123** [comment]
@BenWilson2 

Would you mind wait my PR https://github.com/mlflow/mlflow/pull/6530 merged and then update your PR ?

My PR has removed the TF estimator autologging. So it conflicts with your PR.

**Comment by BenWilson2** [comment]
> @BenWilson2
> 
> Would you mind wait my PR #6530 merged and then update your PR ?
> 
> My PR has removed the TF estimator autologging. So it conflicts with your PR.

No problem at all! I'll resubmit this PR after I branch off of your merged changes

File: tests/tensorflow/test_tensorflow2_autolog.py [hunk 1]
```diff
@@ -1569,20 +1569,6 @@ def test_keras_autolog_infers_model_signature_correctly_with_keras_sequence(
...
-):
-    mlflow.autolog()
-    initial_model = create_tf_keras_model()
-    initial_model.fit(keras_data_gen_sequence)
-
-    mlmodel_path = mlflow.artifacts.download_artifacts(
-        f"runs:/{mlflow.last_active_run().info.run_id}/model/MLmodel"
-    )
-    mlmodel_contents = yaml.safe_load(open(mlmodel_path, "r"))
-    assert "signature" not in mlmodel_contents, mlmodel_contents.keys()
```
Comment (dbczumar) [inline tests/tensorflow/test_tensorflow2_autolog.py:1583]: Can we convert this to a `test_keras_autolog_logs_model_signature` test case?

**Review by dbczumar — APPROVED** [review]
LGTM with one small comment (https://github.com/mlflow/mlflow/pull/6678/files#r961213329). Thanks @BenWilson2 !

File: tests/tensorflow/test_tensorflow2_autolog.py (hunk 1 above)
Comment (BenWilson2) [reply tests/tensorflow/test_tensorflow2_autolog.py]: added with signature check and signature content validation

**Pull request merged**